Sources/ComposableArchitecture/SwiftUI/ForEachStore.swift

```diff
@@ -22,27 +22,18 @@ where Data: Collection, ID: Hashable, Content: View {
     Data == [EachState],
     EachContent: View,
     Content == WithViewStore<
-      Data, (Data.Index, EachAction),
-      ForEach<ContiguousArray<(Data.Index, EachState)>, ID, EachContent>
+      [ID], (Data.Index, EachAction),
+      ForEach<[(offset: Int, element: ID)], ID, EachContent>
     >
   {
-    self.data = ViewStore(store, removeDuplicates: { _, _ in false }).state
+    self.data = store.state.value
     self.content = {
-      WithViewStore(
-        store,
-        removeDuplicates: { lhs, rhs in
-          guard lhs.count == rhs.count else { return false }
-          return zip(lhs, rhs).allSatisfy { $0[keyPath: id] == $1[keyPath: id] }
-        }
-      ) { viewStore in
-        ForEach(
-          ContiguousArray(zip(viewStore.indices, viewStore.state)),
-          id: (\(Data.Index, EachState).1).appending(path: id)
-        ) { index, element in
+      WithViewStore(store.scope(state: { $0.map { $0[keyPath: id] } })) { viewStore in
```

Comment:
We can rely on the `ID`s here to handle the `removeDuplicates` logic automatically.